bot/src/main/java/edu/java/bot/configuration/ApplicationConfiguration.java

```diff
@@ -1,45 +1,51 @@
 package edu.java.bot.configuration;
 
-import edu.java.bot.model.User;
-import edu.java.bot.model.link.Link;
+import com.pengrad.telegrambot.TelegramBot;
+import edu.java.bot.model.command.Command;
+import edu.java.bot.model.command.impl.HelpCommand;
+import edu.java.bot.model.command.impl.ListCommand;
+import edu.java.bot.model.command.impl.ResetCommand;
+import edu.java.bot.model.command.impl.StartCommand;
+import edu.java.bot.model.command.impl.TrackCommand;
+import edu.java.bot.model.command.impl.UntrackCommand;
 import edu.java.bot.model.link.parser.GithubParser;
 import edu.java.bot.model.link.parser.LinkParser;
 import edu.java.bot.model.link.parser.StackOverflowParser;
 import edu.java.bot.service.BotService;
-import java.net.URI;
-import lombok.AllArgsConstructor;
-import org.springframework.beans.factory.config.BeanDefinition;
-import org.springframework.boot.SpringBootConfiguration;
+import java.util.HashMap;
+import java.util.Map;
 import org.springframework.context.ApplicationContext;
 import org.springframework.context.annotation.Bean;
-import org.springframework.context.annotation.Scope;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Primary;
 
-@SpringBootConfiguration
-@AllArgsConstructor
+@Configuration
 public class ApplicationConfiguration {
-    private final ApplicationContext context;
-
     @Bean
-    @Scope(BeanDefinition.SCOPE_PROTOTYPE)
-    public User user(long id) {
-        return new User(
-            id,
-            context.getBean(BotService.class).getBot()
+    @Primary
+    public LinkParser parserChain(ApplicationContext context) {
+        return LinkParser.link(
+            context.getBean(GithubParser.class),
+            context.getBean(StackOverflowParser.class)
         );
     }
 
     @Bean
-    @Scope(BeanDefinition.SCOPE_PROTOTYPE)
-    public Link link(URI uri) {
-        return new Link(uri);
+    public Map<String, Command> commandMap(ApplicationContext context) {
```

Comment:
Сделай метод внутри команды, которое будет возвращать её имя,  и получай на вход сразу команды, а не контекст
```suggestion
    public Map<String, Command> commandMap(List<Command>  commands) {
```